crates/pathfinder/src/state/sync/l1.rs

```diff
@@ -29,29 +30,31 @@ where
     T: EthereumApi + Clone,
 {
     let L1SyncContext {
-        ethereum,
+        mut ethereum,
         chain: _,
         core_address,
         poll_interval,
     } = context;
 
-    let mut previous = EthereumStateUpdate::default();
-
-    loop {
-        let state_update = Retry::exponential(
-            || async { ethereum.get_starknet_state(&core_address).await },
-            NonZeroU64::new(1).unwrap(),
-        )
-        .factor(NonZeroU64::new(2).unwrap())
-        .max_delay(poll_interval / 2)
-        .when(|_| true)
+    // Listen for state updates and send them to the event channel
+    let tx_event = std::sync::Arc::new(tx_event);
+    ethereum
+        .listen(&core_address, poll_interval, move |event| {
+            let tx_event = tx_event.clone();
+            async move {
+                match event {
+                    EthereumEvent::StateUpdate(state_update) => {
+                        info!("State update: {:?}", state_update);
```

Comment:
Please move this to `trace!()` instead. We should avoid littering default (info-level) logs with this.